File: app/src/main/java/com/wnet/imageconvertor/util/ImageConverter.java
package com.wnet.imageconvertor.util;

//public class ImageConverter extends AsyncTask<Void, Void, String> {
//
//    private MainActivity activity;
//    private String type;
//    private Bitmap imageBitmap;
//    private int quality;
//
//    public ImageConverter(MainActivity activity, String type, int quality, Bitmap imageBitmap) {
//        this.activity = activity;
//        this.type = type;
//        this.quality = quality;
//        this.imageBitmap = imageBitmap;
//    }
//
//    @Override
//    protected String doInBackground(Void... params) {
//        try {
//            String filename = generateUniqueImageFileName();
////            FileOutputStream out = new FileOutputStream(filename);
//            if(type.equals("JPEG")){
//                filename = filename+".jpg";
//            }else if(type.equals("PNG")){
//                filename = filename+".png";
//            }else if(type.equals("WEBP")){
//                filename = filename+".webp";
//            }
//            FileOutputStream out = new FileOutputStream(new File(android.os.Environment.getExternalStorageDirectory(), filename));
//            if(type.equals("JPEG")){
//                imageBitmap.compress(Bitmap.CompressFormat.JPEG, quality, out); //100-best quality
//            }else if(type.equals("PNG")){
//                imageBitmap.compress(Bitmap.CompressFormat.PNG, quality, out); //100-best quality
//            }else if(type.equals("WEBP")){
//                imageBitmap.compress(Bitmap.CompressFormat.WEBP, quality, out); //100-best quality
//            }
//            out.close();
//            return filename;
//        } catch (Exception e) {
//            e.printStackTrace();
//            return null;
//        }
//    }
//
//
//
//    @Override
//    protected void onPostExecute(String filePath) {
//        activity.convertedImage(filePath);
//    }
//}

File: app/src/main/java/com/wnet/imageconvertor/util/ImageConverter.kt
package com.wnet.imageconvertor.util

import android.graphics.Bitmap
import android.os.AsyncTask

import com.wnet.imageconvertor.ui.homescreen.HomeFragment

import java.io.File
import java.io.FileOutputStream

import com.wnet.imageconvertor.util.Utils.generateUniqueImageFileName

class ImageConverter(val context: HomeFragment, private val type: String, private val quality: Int, private val imageBitmap: Bitmap) : AsyncTask<Void, Void, String>() {

    override fun doInBackground(vararg params: Void): String? {
        try {
            var filename = generateUniqueImageFileName()
            //            FileOutputStream out = new FileOutputStream(filename);
            if (type == "JPEG") {
                filename = "$filename.jpg"
            } else if (type == "PNG") {
                filename = "$filename.png"
            } else if (type == "WEBP") {
                filename = "$filename.webp"
            }
            val out = FileOutputStream(File(android.os.Environment.getExternalStorageDirectory(), filename))
            if (type == "JPEG") {
                imageBitmap.compress(Bitmap.CompressFormat.JPEG, quality, out) //100-best quality
            } else if (type == "PNG") {
                imageBitmap.compress(Bitmap.CompressFormat.PNG, quality, out) //100-best quality
            } else if (type == "WEBP") {
                imageBitmap.compress(Bitmap.CompressFormat.WEBP, quality, out) //100-best quality
            }
            out.close()
            return filename
        } catch (e: Exception) {
            e.printStackTrace()
            return null
        }

    }


    override fun onPostExecute(filePath: String) {
        context.convertedImage(filePath)
    }


    interface ImageConvertorCallback {
        fun imageConvertorCallback(filePath: String)
    }
}
